ElectronicObserver/Window/Wpf/ShipTrainingPlanner/ShipTrainingPlanViewerViewModel.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using CommunityToolkit.Mvvm.DependencyInjection;
+using CommunityToolkit.Mvvm.Input;
+using ElectronicObserver.Behaviors.PersistentColumns;
+using ElectronicObserver.Data;
+using ElectronicObserver.Database;
+using ElectronicObserver.Observer;
+using ElectronicObserver.Resource;
+using ElectronicObserver.ViewModels;
+using ElectronicObserver.Window.Dialog.ShipDataPicker;
+using ElectronicObserverTypes;
+using Jot;
+
+namespace ElectronicObserver.Window.Wpf.ShipTrainingPlanner;
+
+public partial class ShipTrainingPlanViewerViewModel : AnchorableViewModel
+{
+	private ShipDataPickerViewModel PickerViewModel { get; } = new();
+
+	public ObservableCollection<ShipTrainingPlanViewModel> Plans { get; set; } = new();
+
+	private ElectronicObserverContext DatabaseContext { get; } = new();
+
+	public List<ColumnProperties> ColumnProperties { get; set; } = new();
+	public List<SortDescription> SortDescriptions { get; set; } = new();
+
+	public ShipTrainingPlannerTranslationViewModel ShipTrainingPlanner { get; }
+
+	public ShipTrainingPlanViewModel? SelectedPlan { get; set; }
+
+	private Tracker Tracker { get; }
+
+	public ShipTrainingPlanViewerViewModel() : base("ShipTrainingPlanViewer", "ShipTrainingPlanViewer", ImageSourceIcons.GetIcon(IconContent.ItemActionReport))
+	{
+		Tracker = Ioc.Default.GetService<Tracker>()!;
+		ShipTrainingPlanner = Ioc.Default.GetRequiredService<ShipTrainingPlannerTranslationViewModel>();
+
+		Title = ShipTrainingPlanner.ViewTitle;
+		ShipTrainingPlanner.PropertyChanged += (_, _) => Title = ShipTrainingPlanner.ViewTitle;
+
+		SubscribeToApi();
+		StartJotTracking();
+	}
+
+	private void SubscribeToApi()
+	{
+		APIObserver o = APIObserver.Instance;
+
+		o.ApiPort_Port.ResponseReceived += (_, _) => Initialize();
```

Comment:
```cs
public void Initialize()
```
to
```cs
private void Initialize(string apiname, dynamic data)
```

```cs
o.ApiPort_Port.ResponseReceived += (_, _) => Initialize();
```
to
```cs
o.ApiPort_Port.ResponseReceived += Initialize;
```

In the `Initialize` function, remove `if (Plans.Any()) return;` and add `o.ApiPort_Port.ResponseReceived -= Initialize;`.